Makefile

```diff
@@ -63,9 +84,9 @@ $(EXEC): $(OBJS)
 $(OBJS): $(INCL) 
 
 clean:
-	rm -f $(OBJS)
+	rm -f $(OBJS) $(EXEC)
 
 tidy:
 	rm -f $(OBJS) ./$(EXEC)
 
-all:  tidy $(EXEC) clean
```

Comment:
I suspect that the reason why this `tidy` and `clean` existed is because the dependencies are not all captured in the Makefile. So, we should keep an eye on ensuring correct recompiling..